src/main/java/duke/parser/FindCommandParser.java

```diff
@@ -1,4 +1,128 @@
+                        TOKEN_STUDENT_NAME, TOKEN_STUDENTID, TOKEN_STUDENT_COURSE, TOKEN_EMAIL);
+
+        FindCommand.FindLocker findLocker = new FindCommand.FindLocker();
+        FindCommand.FindStudent findStudent = new FindCommand.FindStudent();
+
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_SERIAL)) {
+            SerialNumber serialNumber = ParserCheck.parseSerialNumber(
+                    mapTokensToArguments.getValue(TOKEN_SERIAL).get());
+
+            findLocker.setSerialNumber(serialNumber);
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_ADDRESS)) {
+            Address address = ParserCheck.parseAddress(
+                    mapTokensToArguments.getValue(TOKEN_ADDRESS).get());
+
+            findLocker.setAddress(address);
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_ZONE)) {
+
+            Zone zone = ParserCheck.parseZone(
+                    mapTokensToArguments.getValue(TOKEN_ZONE).get());
+
+            findLocker.setZone(zone);
+
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_CONDITION)) {
+
+            Tag tag = ParserCheck.parseStatus(
+                    mapTokensToArguments.getValue(TOKEN_CONDITION).get());
+
+            findLocker.setTag(tag);
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_STUDENT_NAME)) {
+
+            Name name = ParserCheck.parseName(
+                    mapTokensToArguments.getValue(TOKEN_STUDENT_NAME).get());
+
+            findStudent.setName(name);
+
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_STUDENTID)) {
+
+            MatricNumber matricNumber = ParserCheck.parseMatricNumber(
+                    mapTokensToArguments.getValue(TOKEN_STUDENTID).get());
+
+            findStudent.setMatricNumber(matricNumber);
+
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_STUDENT_COURSE)) {
+
+            Major major = ParserCheck.parseMajor(
+                    mapTokensToArguments.getValue(TOKEN_STUDENT_COURSE).get());
+
+            findStudent.setMajor(major);
+
+        }
+        if (checkTokenPresent(mapTokensToArguments,
+                TOKEN_EMAIL)) {
+
+            Email email = ParserCheck.parseEmail(
+                    mapTokensToArguments.getValue(TOKEN_EMAIL).get());
+
+            findStudent.setEmail(email);
+
+        }
+
+        if (!findLocker.missingFields() && !findStudent.missingFields()) {
+            throw new DukeException(" Invalid command format");
+        }
+
+        return new FindCommand(findLocker, findStudent);
+    }
+
+    private static boolean checkTokenPresent(MapTokensToArguments mapTokensToArguments,
+                                             Token tokens) {
+
+        return mapTokensToArguments.getValue(tokens).isPresent();
+
+    }
+
```

Comment:
if my understanding of find feature is correct. You can literally search for any field you want but can have only like one field at the moment? Also are all the errors handled like null pointer and stuff?